test/TestUtilities/Test.Utility/DependencyResolver/DependencyProvider.cs

```diff
@@ -0,0 +1,169 @@
+// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Threading;
+using System.Threading.Tasks;
+using NuGet.Common;
+using NuGet.Configuration;
+using NuGet.DependencyResolver;
+using NuGet.Frameworks;
+using NuGet.LibraryModel;
+using NuGet.Packaging;
+using NuGet.Packaging.Core;
+using NuGet.Protocol.Core.Types;
+using NuGet.Versioning;
+
+namespace Test.Utility
+{
+    public class DependencyProvider : IRemoteDependencyProvider, IDependencyProvider
```

Comment:
This code is moved from NuGet.DependencyResolver.Core.Tests/RemoteDependencyWalkerTests.cs .